partiql-tests-data/fail/static-analysis/primitives/functions/char_length.ion

```diff
@@ -0,0 +1,156 @@
+wrong_num_function_args::[
+  char_length::[
+    {
+      name:"char_length wrong number of arguments{sql:\"char_length()\"}",
+      statement:"char_length()",
+      assert: {
+        result: StaticAnalysisFail
+      },
+    },
+    {
+      name:"char_length wrong number of arguments{sql:\"char_length('', '')\"}",
+      statement:"char_length('', '')",
+      assert: {
+        result: StaticAnalysisFail
+      },
+    },
+    {
+      name:"char_length wrong number of arguments{sql:\"char_length('', 1)\"}",
+      statement:"char_length('', 1)",
+      assert: {
+        result: StaticAnalysisFail
+      },
+    }
+  ],
+]
+
+invalid_function_args::[
```

Comment:
Function calls with no schema present in STRICT/ERROR typing mode give errors with mistyped function arguments and return `MISSING` in PERMISSIVE/COERCE mode. But since a literal is provided in these tests, a static analysis pass could determine that the query has a mistyped argument and give an error.

From discussion with Almann, we determined that these cases (i.e. mistyped function args with statically-knowable types) should "fail-fast" and not wait till evaluation time. Similar reasoning applies to the other static-analysis tests ported in this PR. This type of static-time error will probably necessitate some RFC.

Reply:
Are you saying the `StaticAnalysisFail` symbol may be a richer element, and the RFC will define this?

Reply:
Not sure what you meant by a "richer element". Could you clarify?

The tests in the `fail/static-analysis` directory define tests that will fail at some stage of static analysis or type-checking. These particular tests with invalid number of arguments and invalid function arguments (known at static-time since they involve literals) are classified as static analysis fail tests in this PR. What I meant by needing an RFC was that this particular behavior isn't part of the current PartiQL spec and we should add an RFC to define the static analysis fail behavior.

Reply:
I can cut a `partiql-spec` issue for static-analysis failures if the above makes sense.